src/DirectGateway.php

```diff
@@ -43,6 +44,16 @@ public function setSimulatorMode($value)
         return $this->setParameter('simulatorMode', $value);
     }
 
+    public function getReferrerId()
+    {
+        return $this->getParameter('referrerId');
+    }
+
+    public function setReferrerId($value)
+    {
+        return $this->setParameter('referrerId', $value);
+    }
+
```

Comment:
Just pick any random driver: Paypal for example get/set Username is here:

https://github.com/thephpleague/omnipay-paypal/blob/master/src/Message/AbstractRequest.php

and here

https://github.com/thephpleague/omnipay-paypal/blob/master/src/ProGateway.php

Should I raise this as a core issue? It is one of those things that I did stumble over when I first started on OmniPay - it did not feel right, but is just one of those things I got used to, and perhaps I should not have rolled over so easily ;-)